RadarSDK/Radar.m

```diff
@@ -711,6 +711,66 @@ + (void)getMatrixFromOrigins:(NSArray<CLLocation *> *_Nonnull)origins
                                         }];
 }
 
++ (void)sendEvent:(NSString *)name
+     withMetadata:(NSDictionary *_Nullable)metadata
+completionHandler:(RadarSendEventCompletionHandler)completionHandler {
+
+    // TODO(jsani): check if permissions are enabled for events that can occur before location permission prompts
+    // if not, send the request with location = NULL
+
+    [self trackOnceWithCompletionHandler:^(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarEvent *> * _Nullable events, RadarUser * _Nullable user) {
+        if (status != RadarStatusSuccess) {
+            if (completionHandler) {
+                [RadarUtils runOnMainThread:^{
+                    completionHandler(status, nil, nil, nil);
+                }];
+            }
+
+            return;
+        }
+
+        [[RadarAPIClient sharedInstance] sendEvent:name withMetadata:metadata user:user completionHandler:^(RadarStatus status, NSDictionary * _Nullable res, RadarEvent * _Nullable event) {
+            if (completionHandler) {
+                NSMutableArray *finalEvents = [NSMutableArray arrayWithArray:events];
+                [finalEvents insertObject:event atIndex:0];
+
+                [RadarUtils runOnMainThread:^{
+                    completionHandler(status, location, finalEvents, user);
```

Comment:
This can either be

```
NSArray *finalEvents = @[events[0]]
 
[RadarUtils runOnMainThread:^{
    completionHandler(status, location, finalEvents, user);
```

or just 

```
[RadarUtils runOnMainThread:^{
    completionHandler(status, location, @[events[0], user);
```

You may also want to check whether `events` is empty before accessing the zeroth element, or else you'll get an `NSRangeException`.